### PR TITLE
Fixed pgsql lastInserteId

### PR DESCRIPTION
**Pull request by lulco**

### CODE REVIEW

**Comment by lulco** [comment]
In PHP 7.0.11 Postgres PDO lastInsertId() was changed: http://php.net/ChangeLog-7.php#7.0.11
It returns "0" instead of false

#hacktoberfest

File: src/Database/Table/Selection.php [hunk 1]
```diff
@@ -821,7 +821,7 @@ public function insert($data)
 				? implode('.', array_map([$this->context->getConnection()->getSupplementalDriver(), 'delimite'], explode('.', $tmp)))
 				: NULL
 		);
-		if ($primaryKey === FALSE) {
+		if ($primaryKey === FALSE || $primaryKey === "0") {
```
Comment (JanTvrdik) [inline src/Database/Table/Selection.php]: the string should be in single quotes, i.e. `$primaryKey === '0'`

Comment (lulco) [inline src/Database/Table/Selection.php]: sure, fixed

**Comment by dg** [comment]
What about simple `if (!$primaryKey)`?

**Comment by JanTvrdik** [comment]
@dg That's less obvious. I think the current solution is better.

**Comment by lulco** [comment]
So? Which alternative will be accepted and merged? I am OK with all of them, just need this fix.

**Comment by dg** [comment]
I dont like fix for specific DB in common class.

**Comment by lulco** [comment]
Well, I just realize that it is not just pgsql problem. `$this->context->getInsertId()` returns '0' also with MySql DB if you have non-autoincrement primary key. And in MySql DB it is happening also in previous PHP versions.

This part of code is executed:
```
$row = $this->createSelectionInstance()
			->select('*')
			->wherePrimary($primaryKey)
			->fetch();
```
And returns `FALSE`, because the executed query is: `... WHERE primary_key = '0' ...`

In this query is the only driver-specific problem with pgsql uuid type, because it throws exception:
`Nette\Database\DriverException: SQLSTATE[22P02]: Invalid text representation: 7 ERROR:  invalid input syntax for uuid: "0"`. The exception in query that shouldn't be executed at all. Definitelly not with $primaryKey = '0'.

**Comment by dg** [comment]
It seems that it is '' in mssql

**Comment by hrach** [comment]
This seems to be a wrong fix; personally, I have a db with table with primay value `0` (yes, big bad design, but that's the current state) . The getLastIndsertedId() should be fixed in the proper layer (Driver/Connection), shouldn't be?

**Comment by dg** [comment]
So since PHP 7.0.11 can be '0' missing primary and correct primary? I don't know how to fix it. The only solution I can see to not support primary value '0'.

**Comment by hrach** [comment]
You can definitely in MySQL insert '0' into a PK column. I'm ok with not-supporting the PK '0', but the unification should may be done in PgSQLDriver.

**Comment by dg** [comment]
MySQL returns '0' when primary key is missing. It seems that condition was wrong.

**Pull request closed without merging**